server/src/immich/api-v1/asset/asset-repository.ts

```diff
@@ -129,21 +129,28 @@ export class AssetRepository implements IAssetRepository {
         isVisible: true,
         isFavorite: dto.isFavorite,
         isArchived: dto.isArchived,
-        updatedAt: dto.updatedAfter ? MoreThan(dto.updatedAfter) : undefined,
+        updatedAt: this.dateFilter(dto.updatedAfter, dto.updatedBefore),
       },
       relations: {
         exifInfo: true,
         tags: true,
         stack: true,
       },
       skip: dto.skip || 0,
+      take: dto.take,
       order: {
         fileCreatedAt: 'DESC',
       },
       withDeleted: true,
     });
   }
 
+  private dateFilter(after?: Date, before?: Date): FindOperator<Date> | undefined {
+    if (before && after) return Between(after, before);
+    if (before) return LessThan(before);
+    if (after) return MoreThan(after);
+  }
+
```

Comment:
We already have this exact code. It is called OptionalBetween. I think it can be imported from infra utils